### PR TITLE
python/metapackages/setup.py: improve package conflict detection

### PR DESCRIPTION
**Pull request by mitchdz**
### Description

- improve installed package detection during `pip install`
- re-enable conflicting package detection text

This code specifically checks the following packages: `cuda-quantum`, `cuda-quantum-cu11`, `cuda-quantum-cu12` during the package installation phase of `pip install`.

For example, with `cuda-quantum-cu11` installed and `cuda12` libraries installed, attempting to `pip install cudaq` will give:
```
Exception: You have a conflicting CUDA-Q version installed.Please remove the following package(s): cuda-quantum-cu11
```

### CODE REVIEW

File: python/metapackages/setup.py [hunk 1]
```diff
@@ -116,22 +117,75 @@ def _get_cuda_version() -> Optional[int]:
...
+    # For some reason, that does not work during pip install, which I'm
+    # not clear about for now. This is worked around by getting the site
+    # packages and searching for the files manually.
+
+    def _check_in_directory(directory, package_name):
+        """
+        Helper function to check if a package exists in a given directory.
+        """
+        search_pattern = os.path.join(directory, f"{package_name}-*.dist-info")
+        matches = glob.glob(search_pattern)
```
[the remark this answers is not in the document]
Comment (mitchdz) [reply python/metapackages/setup.py:135]: I didn't find a great file to specifically look for that all installed packages would have. The consistent one was `${package_name}-${version}.dist-info` which contains the metadata for the package.

File: python/metapackages/setup.py [hunk 2]
```diff
@@ -116,22 +117,75 @@ def _get_cuda_version() -> Optional[int]:
...
+    # cuda_quantum_cu11.libs
+    # cuda_quantum_cu12-0.9.0.dist-info
+    # cuda_quantum_cu12.libs
+    normalized_package_name = _replace_hyphens_with_underscores(package_name)
+
+    # Check in user site-packages
+    user_site_packages = site.getusersitepackages()
+    if os.path.exists(user_site_packages):
+        _log(f"User site-packages directory: {user_site_packages}")
+        if _check_in_directory(user_site_packages, normalized_package_name):
```
[the remark this answers is not in the document]
Comment (mitchdz) [reply python/metapackages/setup.py]: Fixed this, thanks.

**Comment by mitchdz** [comment]
FYI - I added a short one line commit to fix the cuda-quantum package README.

**Review by bettinaheim — APPROVED** [review]

File: python/metapackages/setup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mitchdz) [reply python/metapackages/setup.py:135]: Resolving this conversation, please let me know if you would like to reopen :)

**Pull request merged**